### DISCUSSION TITLE
How to fix padding in glyphs?

### DISCUSSION THREAD

**Discussion by augustocdias** [ryanoasis/nerd-fonts#874] [Q&A] [answered]
I'm trying to patch codicons into Firacode and I'm having some issues. It is adding a bottom padding to the glyphs.

![image](https://user-images.githubusercontent.com/1080789/183072165-a592e113-ab50-4284-aa78-2ab6f31b0e8d.png)

Is there a flag or a setting I can set to adjust this and fix it?

This is how I patched it:

```
fontforge -script font-patcher  /Users/augusto/Library/Fonts/Fira\ Code\ Retina\ Nerd\ Font\ Complete\ Mono.otf --custom codicon.ttf 
```

I installed the font from brew.

**Comment by Finii** [comment, marked as answer]
The padding (i.e. vertical alignment) should not be wrong. Esp if you use a pre-patched font the algorithm should calculate the cell size and midline identical to the previous run (the run that created the font you fetched via brew).
This is puzzling.

And to answer the concrete question: There is no flag or option to manipulate the vertical alignment.

As solution you might consider using `v2.2.0 RC` instead, as that is prepatched with _Codicons 1.5_. Current seems to be _1.9_.

I guess in your screenshot the 'wrong' icons are the blue box and the green wrench. That would be codepoints `EA8C` and `EB65`.

**Reply by Finii** [reply]
https://github.com/ryanoasis/nerd-fonts/issues/877

**Reply by augustocdias** [reply]
Yes. The wrong icons are the blue and green ones. Most icons from codicons are having this issue. 

I'll try the new version. Tks

**Reply by Finii** [reply]
Found the reason for the behavior you experienced and it is ... not-a-bug ;-)
See explanation in #877 

`--custom` is working ... strangely anyhow. Not sure what people use it for normally.

**Reply by augustocdias** [reply]
Is there a way to pass parameters to the script to "fix" this behavior?

**Reply by Finii** [reply]
No.
Otoh codicons is already in Nerd Fonts, so there should be no reason to do what you tried?

The only possibility right now is editing the script, if it is possible for you,

You have to add valign-center. If you want to scale add stretch-pa:

```diff
         CUSTOM_ATTR = {
             # 'pa' == preserve aspect ratio
-            'default': {'align': 'c', 'valign': '', 'stretch': '', 'params': {}}
+            'default': {'align': 'c', 'valign': 'c', 'stretch': 'pa', 'params': {}}
         }
```

If you enable scaling, this might be what you want, or not. For the scaling there are even more rules possible and it can become more complicated.

Edit: We add the codicons just like that (just valign-c stretch-pa). No other rules are in effect.

**Reply by augustocdias** [reply]
Are they the same codes? I remember I posted this because I couldn't use them... I had just copied them from the check here: https://github.com/mortepau/codicons.nvim#how-to-patch-fonts and since it wasn't working I tried patching.

**Reply by Finii** [reply]
The icons are at their native codepoints...:

![image](https://user-images.githubusercontent.com/16012374/211527925-a1fdf6c2-185f-416f-a642-7e885da06206.png)

(Font shown is `master`; and I do not know which codicons version we have currently and what their latest release is.)

(I'm not sure I understand the question right.)

**Reply by augustocdias** [reply]
I'll try again. Maybe I had an outdated version
